src/core/font_renderer.js

```diff
@@ -749,22 +741,66 @@ const NOOP = [];
 class Commands {
   cmds = [];
 
+  transformStack = [];
+
+  currentTransform = [1, 0, 0, 1, 0, 0];
+
   add(cmd, args) {
     if (args) {
       if (!isNumberArray(args, null)) {
         warn(
           `Commands.add - "${cmd}" has at least one non-number arg: "${args}".`
         );
```

Comment:
I think this check is useless since the numbers are only coming from the binary font data.

Reply:
> I think this check is useless since the numbers are only coming from the binary font data.

*Agreed, but please see below for additional details.*

The `fontMatrix` used in this file can also *indirectly* come from the /FontMatrix entry in the font-dictionary, which was the source of the security problem in https://bugzilla.mozilla.org/show_bug.cgi?id=1893645.

However, please note that that's no longer a concern since we're now validating that entry *properly* when looking it up:
https://github.com/mozilla/pdf.js/blob/3f1d07a35e8641f0b85b1262eabfb399253ff0b4/src/core/evaluator.js#L4483-L4486

(After that security bug, I also went through all the `src/core/` code and added validation everywhere that we're looking up Array-data from a dictionary.)